scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
+        try {
+            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
+            ps = con.prepareStatement(loginUserQuery);
+            ps.setString(1, email);
+            ps.setString(2, password);
+            ps.setString(3, userType);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                user = new User();
+                user.setFirstName(rs.getString("firstName"));
+                user.setLastName(rs.getString("lastName"));
+                user.setPhone(rs.getLong("phone"));
+                user.setEmailId(email);
+                user.setPassword(password);
+                session.setAttribute(role.toString(), user.getEmailId());
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return user;
+    }
+
+    @Override
+    public boolean isLoggedIn(UserRole role, HttpSession session) {
+        if (role == null)
+            role = UserRole.CUSTOMER;
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    @Override
+    public boolean logout(HttpSession session) {
+        session.removeAttribute(UserRole.CUSTOMER.toString());
+        session.removeAttribute(UserRole.SELLER.toString());
+        session.invalidate();
+        return true;
+    }
+
+    @Override
+    public String register(UserRole role, User user) throws StoreException {
+        String responseMessage = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(registerUserQuery);
+            ps.setString(1, user.getEmailId());
+            ps.setString(2, user.getPassword());
+            ps.setString(3, user.getFirstName());
+            ps.setString(4, user.getLastName());
+            ps.setString(5, user.getAddress());
+            ps.setLong(6, user.getPhone());
+            ps.setString(7, user.getEmailId());
+            int userType = UserRole.SELLER.equals(role) ? 1 : 2;
+            ps.setInt(8, userType);
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseMessage = ResponseCode.SUCCESS.name();
+                ;
+            }
+        } catch (Exception e) {
+            responseMessage += " : " + e.getMessage();
+            if (responseMessage.contains("Duplicate"))
+                responseMessage = "User already registered with this email !!";
+            e.printStackTrace();
+        }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Improve exception handling in register method</b></div>
<div id="fix">

Similar to login method, consider proper exception handling here instead of concatenating error message and printing stack trace
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -89,7 +89,6 @@
 - } catch (Exception e) {
 -     responseMessage += " : " + e.getMessage();
 -     if (responseMessage.contains("Duplicate"))
 -         responseMessage = "User already registered with this email !!";
 -     e.printStackTrace();
 + } catch (Exception e) {
 +     String message = e.getMessage().contains("Duplicate") ? "User already registered with this email" : "Error registering user";
 +     throw new StoreException(message);
  }
```


```suggestion
 } catch (Exception e) {
     String message = e.getMessage().contains("Duplicate") ? "User already registered with this email" : "Error registering user";
     throw new StoreException(message);
}
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2527668117>#282fd2</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged